modules/apps/frontend-js/frontend-js-react-web/src/main/resources/META-INF/resources/js/hooks/useControlledState.ts

```diff
@@ -0,0 +1,49 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2025 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {useCallback, useState} from 'react';
+
+export type InternalDispatch<Value> =
+	| ((value: Value) => void)
+	| ((value?: Value) => void)
+	| ((value: Value | (() => Value)) => void)
+	| ((value?: Value | (() => Value)) => void)
+	| React.Dispatch<React.SetStateAction<Value>>;
+
+type Props<Value> = {
+	defaultValue?: Value | (() => Value);
+	onChange?: InternalDispatch<Value>;
+	value?: Value;
+};
+
+export default function useControlledState<Value>({
```

Comment:
Is there a reason not to use the useControlledState hook from Clay/Shared?

Reply:
I wasn't aware of it being internal only, but you would probably have more insight into those past decisions than I would. Personally, I don't see a problem in using it from `shared`. Nothing about that name sounds like an internal only package to me. If we shouldn't use it, then we should also refactor the new ItemSelector to not use it.